src/parser/jobs/dnc/modules/DirtyDancing.tsx

```diff
@@ -0,0 +1,298 @@
+		if (actualFinish) {
+			if (TECHNICAL_FINISHES.includes(actualFinish.ability.guid)) {
+				expectedFinish =  ACTIONS.QUADRUPLE_TECHNICAL_FINISH.id
+			} else if (STANDARD_FINISHES.includes(actualFinish.ability.guid)) {
+				expectedFinish = ACTIONS.DOUBLE_STANDARD_FINISH.id
+			}
+		}
+		return expectedFinish
+	}
+
+	constructor(start: number) {
+		this.start = start
+		this.dancing = true
+	}
+}
+
+export default class DirtyDancing extends Module {
+	static handle = 'dirtydancing'
+	static title = t('dnc.dirty-dancing.title')`Dance Issues`
+
+	@dependency private checklist!: CheckList
+	@dependency private suggestions!: Suggestions
+	@dependency private invuln!: Invulnerability
+	@dependency private combatants!: Combatants
+	@dependency private timeline!: Timeline
+
+	private danceHistory: Dance[] = []
+	private missedDances = 0
+	private dirtyDances = 0
+	private footlooseDances = 0
+
+	protected init() {
+		this.addHook('cast', {by: 'player', abilityId: STEPS}, this.beginDance)
+		this.addHook('cast', {by: 'player'}, this.continueDance)
+		this.addHook('cast', {by: 'player', abilityId: FINISHES}, this.finishDance)
+		this.addHook('damage', {by: 'player', abilityId: FINISHES}, this.resolveDance)
+		this.addHook('complete', this.onComplete)
+	}
+
+	private addDanceToHistory(event: CastEvent): Dance {
+		const newDance = new Dance(event.timestamp)
+		newDance.rotation.push(event)
+		this.danceHistory.push(newDance)
+
+		return newDance
+	}
+
+	private beginDance(event: CastEvent) {
+		this.addDanceToHistory(event)
+	}
+
+	private get lastDance(): Dance | undefined {
+		return _.last(this.danceHistory)
+	}
+
+	private continueDance(event: CastEvent) {
+		if (!STEPS.includes(event.ability.guid) && !FINISHES.includes(event.ability.guid)) {
+			const dance = this.lastDance
+			if (dance && dance.dancing) {
+				dance.rotation.push(event)
+			}
+		}
+	}
+
+	private finishDance(event: CastEvent) {
+		let dance = this.lastDance
+		if (dance && dance.dancing) {
+			dance.rotation.push(event)
+		} else {
+			dance = this.addDanceToHistory(event)
+		}
+		dance.dancing = false
+	}
+
+	private resolveDance(event: DamageEvent) {
+		const dance = this.lastDance
+		if (dance && !dance.resolved) {
+
+			const finisher = dance.rotation[dance.rotation.length-1]
+			dance.end = finisher.timestamp
+
+			// Count dance as dirty if we didn't get the expected finisher
+			if (finisher.ability.guid !== dance.expectedFinishId) {
+				dance.dirty = true
+			}
+			// If the finisher didn't hit anything, and something could've been, ding it.
+			// Don't gripe if the boss is invuln, there is use-case for finishing during the downtime
+			if (event.amount === 0 && !this.invuln.isInvulnerable('all', finisher.timestamp)) {
+				dance.missed = true
+			}
+			// Dancer messed up if more step actions were recorded than we expected
+			const actualCount = dance.rotation.filter(step => DANCE_MOVES.includes(step.ability.guid)).length
+			let expectedCount = 0
```

Comment:
this isn't doing much; can just `const` is and set it with the value on the next line.